plugins/inputs/huebridge/huebridge.go

```diff
@@ -0,0 +1,70 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package huebridge
+
+import (
+	_ "embed"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type RemoteClientConfig struct {
+	RemoteClientId     string `toml:"remote_client_id"`
+	RemoteClientSecret string `toml:"remote_client_secret"`
+	RemoteCallbackUrl  string `toml:"remote_callback_url"`
+	RemoteTokenDir     string `toml:"remote_token_dir"`
+}
+
+type HueBridge struct {
+	Bridges         []string          `toml:"bridges"`
+	RoomAssignments map[string]string `toml:"room_assignments"`
+	Timeout         config.Duration   `toml:"timeout"`
+	Log             telegraf.Logger   `toml:"-"`
+	RemoteClientConfig
+	tls.ClientConfig
+
+	configuredBridges []*bridge
```

Comment:
How about using a bit shorter names?
```suggestion
	bridges []*bridge
```
Not a must, only a suggestion...

Reply:
Adapted via 62d8804ee8396401842e4c93008d15d441651751